database/database.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package database
+
+import (
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/pkg/errors"
+)
+
+const (
+	LatestSeenBlockKey = "latestSeenBlock"
+)
+
+var (
+	ErrKeyNotFound              = errors.New("key not found")
+	ErrChainNotFound            = errors.New("no database for chain")
+	ErrDatabaseMisconfiguration = errors.New("database misconfiguration")
+)
+
+// RelayerDatabase is a key-value store for relayer state, with each chainID maintaining its own state
+type RelayerDatabase interface {
+	Get(chainID ids.ID, key []byte) ([]byte, error)
```

Comment:
Why do we return []byte instead of string directly? When we write in to the file, we convert []byte to string. Also, when we use the result, we convert []byte to string.

Reply:
We discussed this offline, but for visibility, I chose `[]byte` in the interface so that the keys and values could be as general as possible. The underlying JSON databse uses strings so that the resulting file is human readable.